server/src/main/java/org/opensearch/action/admin/cluster/remotestore/stats/RemoteStoreStats.java

```diff
@@ -0,0 +1,108 @@
+        builder.field(Fields.STARTED, remoteSegmentUploadShardStats.totalUploadsStarted)
+            .field(Fields.SUCCEEDED, remoteSegmentUploadShardStats.totalUploadsSucceeded)
+            .field(Fields.FAILED, remoteSegmentUploadShardStats.totalUploadsFailed);
+        builder.endObject();
+
+        builder.startObject(Fields.TOTAL_UPLOADS_IN_BYTES);
+        builder.field(Fields.STARTED, remoteSegmentUploadShardStats.uploadBytesStarted)
+            .field(Fields.SUCCEEDED, remoteSegmentUploadShardStats.uploadBytesSucceeded)
+            .field(Fields.FAILED, remoteSegmentUploadShardStats.uploadBytesFailed);
+        builder.endObject();
+        builder.startObject(Fields.REMOTE_REFRESH_SIZE_IN_BYTES);
+        builder.field(Fields.LAST_SUCCESSFUL, remoteSegmentUploadShardStats.lastSuccessfulRemoteRefreshBytes);
+        builder.field(Fields.MOVING_AVG, remoteSegmentUploadShardStats.uploadBytesMovingAverage);
+        builder.endObject();
+
+        builder.startObject(Fields.UPLOAD_LATENCY_IN_BYTES_PER_SEC);
+        builder.field(Fields.MOVING_AVG, remoteSegmentUploadShardStats.uploadBytesPerSecMovingAverage);
+        builder.endObject();
+        builder.startObject(Fields.REMOTE_REFRESH_LATENCY_IN_NANOS);
+        builder.field(Fields.MOVING_AVG, remoteSegmentUploadShardStats.uploadTimeMovingAverage);
+        builder.endObject();
+        builder.endObject();
+
+        return builder;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeOptionalWriteable(remoteSegmentUploadShardStats);
+    }
+
+    static final class Fields {
+        static final String SHARD_ID = "shard_id";
+        static final String LOCAL_REFRESH_TIMESTAMP = "local_refresh_timestamp_in_millis";
+        static final String LOCAL_REFRESH_CUMULATIVE_COUNT = "local_refresh_cumulative_count";
+        static final String REMOTE_REFRESH_TIMESTAMP = "remote_refresh_timestamp_in_millis";
+        static final String REMOTE_REFRESH_CUMULATIVE_COUNT = "remote_refresh_cumulative_count";
+        static final String BYTES_LAG = "bytes_lag";
+        static final String REJECTION_COUNT = "rejection_count";
+        static final String CONSECUTIVE_FAILURE_COUNT = "consecutive_failure_count";
+        static final String TOTAL_REMOTE_REFRESH = "total_remote_refresh";
+        static final String TOTAL_UPLOADS_IN_BYTES = "total_uploads_in_bytes";
+        static final String REMOTE_REFRESH_SIZE_IN_BYTES = "remote_refresh_size_in_bytes";
+        static final String UPLOAD_LATENCY_IN_BYTES_PER_SEC = "upload_latency_in_bytes_per_sec";
+        static final String REMOTE_REFRESH_LATENCY_IN_NANOS = "remote_refresh_latency_in_nanos";
+        static final String STARTED = "started";
+        static final String SUCCEEDED = "succeeded";
+        static final String FAILED = "failed";
+        static final String MOVING_AVG = "moving_avg";
+        static final String LAST_SUCCESSFUL = "last_successful";
```

Comment:
lets make sure we create an issue in documentation repo to add clear description for each of the fields. For eg - Fields.LOCAL_REFRESH_TIMESTAMP is the timestamp of the latest local refresh.

Reply:
The fields that are not immediately obvious should probably be documented here in the code now while the context is fresh.